R/map_vis.R

```diff
@@ -294,3 +294,408 @@ plot_watershed <- function(data_dir,
+           `Cooling Technology` = as.factor(cooling_tech))
+
+  power_plants_USA[watersheds_city, ] -> power_plants_city
+
+  # read NID point file and select only Flood Control Dams (C = Flood Control)
+  import_shapefile(paste0(data_dir, dams_file_path)) %>%
+    #subset(grepl("C", Purposes)) %>%
+    as_Spatial() -> flood_control_dams
+
+  # read crop raster for US
+  import_raster(paste0(data_dir, crop_file_path)) ->
+    cropcover_USA
+  # get crop classes and color scheme
+
+  raster::crs(cropcover_USA) <- raster::crs(watersheds_city)
+
+  # read crop table
+  foreign::read.dbf(paste0(data_dir, crop_attributes)) ->crop_cover_levels
+
+  crop_cover_levels %>%
+    filter(Class_Name != "") %>%
+    mutate(color = rgb(Red, Green, Blue, maxColorValue = 255)) %>%
+    dplyr::select(Value, cropclass = Class_Name, color) %>%
+    mutate(color = if_else(Value == 0, "#ffffff", color)) %>%
+    mutate(cropclass = as.character(cropclass)) -> cropclasses
+
+  cropclasses %>%
+    dplyr::rename(ID = Value) %>%
+    select(ID, cropclass) %>%
+    mutate(cropclass = if_else(ID == 131, "Barren_", cropclass),
+           cropclass = if_else(ID == 152, "Shrubland_", cropclass)) -> crop_tbl
+  crop_tbl$ID <- as.integer(crop_tbl$ID)
+  crop_tbl$cropclass <- as.factor(crop_tbl$cropclass)
+  levels(cropcover_USA) <- crop_tbl
+
+  # make color palette
+  setNames(cropclasses$color, cropclasses$cropclass) -> crop_palette
+
+  # mask to required city
+  cropcover_USA %>%
+    mask_raster_to_polygon(watersheds_city) ->
+    cropcover_city_watersheds_agg
+  cropcover_city_watersheds_agg@legend@colortable <- logical(0)
+
+  rm(cropcover_USA)
+
+  # get the coordinate system of the crop raster and transform watershed
+  r_crs <- st_crs(projection(cropcover_city_watersheds_agg))
+  watersheds_city_trans <- st_as_sf(watersheds_city) %>% st_transform(crs = r_crs)
+  state_shape_trans <- st_as_sf(state_shape) %>% st_transform(crs = r_crs)
+
+  # create extents to define bounding boxes of maps
+  watersheds_city_trans %>% extent() -> ws_extent
+  state_shape_trans %>% extent() -> state_extent
+  extent_y_diff <- (ws_extent@ymax - ws_extent@ymin) * 0.05
+  extent_x_diff <- (ws_extent@xmax - ws_extent@xmin) * 0.05
+
+  extent(
+    ws_extent@xmin - extent_x_diff,
+    ws_extent@xmax + extent_x_diff,
+    ws_extent@ymin - extent_y_diff,
+    ws_extent@ymax + extent_y_diff
+  ) -> map_extent
+
+  extent(
+    min(ws_extent@xmin, state_extent@xmin),
+    max(ws_extent@xmax, state_extent@xmax),
+    min(ws_extent@ymin, state_extent@ymin),
+    max(ws_extent@ymax, state_extent@ymax)
+  ) -> state_ws_extent
+
+  map.p <- rasterToPoints(cropcover_city_watersheds_agg)
+  df <- data.frame(map.p)
+  colnames(df) <- c("Longitude", "Latitude", "Value")
+  left_join(df, cropclasses, by = "Value") -> df2
+
+  flood_control_dams %>% st_as_sf() -> flood_control_sf
+  watersheds_city %>% st_as_sf() -> watershed_sf
+  flood_control_sf[watershed_sf, ] -> select_dams
```

Comment:
I get this error message here:
```
Error in st_geos_binop("intersects", x, y, sparse = sparse, prepared = prepared) : 
  st_crs(x) == st_crs(y) is not TRUE
```

Reply:
still getting this error... any ideas? Is this running clean for you if you clean your environment and attempt on a few different cities?

Reply:
Hmm...I am not getting this error, but I can try to add something in that would fix this if I did get the error. I think it is just the a projection problem between the two sf objects.